tests/reset/reset.cpp

```diff
@@ -360,3 +360,266 @@ TEST(Reset, hasResetFromComponentMethod)
+    libcellml::ResetPtr r = std::make_shared<libcellml::Reset>();
+    libcellml::VariablePtr v1 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v2 = std::make_shared<libcellml::Variable>();
+    libcellml::Printer p;
+
+    v1->setName("A");
+    v2->setName("B");
+
+    r->setVariable(v1);
+    r->setTestVariable(v2);
+    r->setOrder(1);
+    c->addReset(r);
+
+    // Test appending of test_value block
+    r->appendTestValue(firstMaths);
+    r->appendTestValue(secondMaths);
+
+    const std::string out = p.printModel(m);
+    EXPECT_EQ(test, out);
+}
+
+TEST(Reset, resetValueSetClear)
+{
+    static const std::string emptyMath =
+        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
+        "</math>\n";
+
+    const std::string in1 =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\">\n"
+        "  <component>\n"
+        "    <reset variable=\"A\" reset_variable=\"B\" order=\"1\"/>\n"
+        "      <reset_value/>\n" // empty reset_value block will be removed on printing
+        "  </component>\n"
+        "</model>\n";
+
+    const std::string test1 =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\">\n"
+        "  <component>\n"
+        "    <reset variable=\"A\" test_variable=\"B\" order=\"1\"/>\n"
+        "  </component>\n"
+        "</model>\n";
+
+    const std::string test2 =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\">\n"
+        "  <component>\n"
+        "    <reset variable=\"A\" test_variable=\"B\" order=\"1\">\n"
+        "      <reset_value>\n"
+        "        <math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
+        "        </math>\n"
+        "      </reset_value>\n"
+        "    </reset>\n"
+        "  </component>\n"
+        "</model>\n";
+
+    libcellml::ModelPtr m = createModelWithComponent();
+    libcellml::ComponentPtr c = m->component(0);
+    libcellml::ResetPtr r = std::make_shared<libcellml::Reset>();
+    libcellml::VariablePtr v1 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v2 = std::make_shared<libcellml::Variable>();
+    libcellml::Printer p;
+
+    v1->setName("A");
+    v2->setName("B");
+
+    r->setVariable(v1);
+    r->setTestVariable(v2);
+    r->setOrder(1);
+    c->addReset(r);
+
+    const std::string out1 = p.printModel(m);
+    EXPECT_EQ(test1, out1);
+
+    // Test setting of reset_value block
+    r->setResetValue(emptyMath);
+    const std::string out2 = p.printModel(m);
+    EXPECT_EQ(test2, out2);
+
+    // Test clearing of reset_value block
+    r->setResetValue("");
+    const std::string out3 = p.printModel(m);
+    EXPECT_EQ(test1, out3);
+}
+
+TEST(Reset, resetValueAppend)
+{
+    static const std::string firstMaths = "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n";
```

Comment:
Could also choose to append two empty maths blocks.

Reply:
Oh, I could have chosen many, many other things ... :)